pkg/agent/manager/cache/cache.go

```diff
@@ -329,11 +325,22 @@ func (c *Cache) UpdateEntries(update *UpdateEntries, checkSVID func(*common.Regi
 			}
 		}
 
-		// Determine if something related to this record changed outside of the
-		// selectors and if so, make sure subscribers for all entry selectors
+		// If any selectors or federated bundles were changed, then make
+		// sure subscribers for the new and extisting entry selector sets
 		// are notified.
-		if federatedBundlesChanged {
-			notifySet.Merge(newEntry.Selectors...)
+		if federatedBundlesChanged || selectorsChanged {
+			if existingEntry != nil {
```

Comment:
One more optimization here. We don't need to notify on the existingEntry set of selectors when `selectorsChanged` is false, since the existingEntry and newEntry selectors will be the same thing.

Reply:
do we need to notify existingEntry when `federatedBundlesChanged` is true? It wasn't doing that before.

Reply:
No, we shouldn't.

Basically the flow should be:
```
if selectorsChanged {
    add newEntry selectors to notify sets
    add existingEntry selectors to notify sets
} else if federatedBundlesChanged {
    add newEntry selectors to notify sets
}```

If the selectors changed, we have to notify workloads that cared about the old set and the new set, since the available SVIDs may have changed. Otherwise, if federated bundles changed, we just notify on the new set, since the old set and new set are equivalent.

Reply:
Thanks, I made the optimization. Do we need the nil check for `newEntry`?

Reply:
Also can 
```
if len(selAdd) > 0 || len(selRem) > 0 || len(fedAdd) > 0 || len(fedRem) > 0 {
```
be condensed to 
```
if federatedBundlesChanged || selectorsChanged {
```

on line 358 of the new version?

Reply:
> Do we need the nil check for newEntry?

No, we don't need to ensure newEntry is not nil. It will always be non-nil.

> be condensed to

Yep! That would be a fine thing to do.

Reply:
ok updated. There are also some newEntry nil checks in diffSelectors() and diffFederatesWith() that could possibly be removed.